#Exploración Inicial de los Datos

####**Autor:** Daniela Zabaleta Benitez
####**Fecha:** 19/08/2025
####**Descripción:**  Notebook para presentar un exploración inicial para entender los datos de la prueba técnica #2, se incluye posibles limpiezas, análisis de datos y visualizaciones básicas.

#Librerias

In [ ]:
import pandas as pd #Manipulacion dataframes

#Cargar los datos

In [ ]:
#Ruta archivos
%cd /content/drive/MyDrive/Colab Notebooks/Prueba Técnica Cientifico de Datos/Prueba tecnica 2

/content/drive/MyDrive/Colab Notebooks/Prueba Técnica Cientifico de Datos/Prueba tecnica 2


In [ ]:
df = pd.read_excel('Datos/sentencias_pasadas.xlsx')
df.head(10)

#  Relevancia Providencia  Tipo Fecha Sentencia  \
0   1   966965.00    T-185/22   NaN      2022-05-31   
1   3   963168.00    T-356/21   NaN      2021-10-15   
2   5   956201.00    T-351/22   NaN      2022-10-07   
3   6   955889.00    T-246/21   NaN      2021-07-29   
4   7   955787.00   T-245A/22   NaN      2022-07-01   
5   8   954029.00    T-190/24   NaN      2024-05-23   
6   9   947406.00    T-394/24   NaN      2024-09-19   
7  10   945053.00    T-155/19   NaN      2019-04-04   
8  11   943156.00    T-339/22   NaN      2022-09-28   
9  14      939.17    T-281/21   NaN      2021-08-23   

                                      Tema - subtema  \
0                                                NaN   
1  ACCIÓN DE TUTELA PARA PROTEGER EL DERECHO A LA...   
2  ACCIÓN DE TUTELA PARA PROTEGER EL DERECHO A LA...   
3  ACCION DE TUTELA PARA PROTEGER EL DERECHO A LA...   
4  ACCION DE TUTELA-Inexistencia de hecho superad...   
5                                                NaN   
6                                                NaN   
7  DERECHO A LA INTIMIDAD-Alcance y contenido DER...   
8  CARENCIA ACTUAL DE OBJETO POR DAÑO CONSUMADO-D...   
9  ACCION DE TUTELA CONTRA PARTICULARES-Solicitud...   

                                            resuelve  \
0  en nombre del pueblo y por mandato de la Const...   
1  en nombre del pueblo y por mandato de la Const...   
2  administrando justicia en nombre del Pueblo, y...   
3  en nombre del pueblo y por mandato de la Const...   
4  en nombre del pueblo y por mandato de la Const...   
5  RESUELVE PRIMERO. CONFIRMAR la decisión profer...   
6  RESUELVE PRIMERO. CONFIRMAR la Sentencia del 1...   
7  en nombre del pueblo y por mandato de la Const...   
8  en nombre del pueblo y por mandato de la Const...   
9  en nombre del pueblo y por mandato de la Const...   

                                            sintesis  
0  En este caso se formula la acción de tutela en...  
1  El peticionario considera que los accionantes ...  
2  El periodista accionante acusa al abogado acci...  
3  Se presenta la acción de tutela en contra de u...  
4  El accionante, actuando en representación de s...  
5  El actor solicitó la protección de los derecho...  
6  La presente acción de tutela fue formulada por...  
7  En el presente caso, se atribuye la vulneració...  
8  La actora adujo que la persona accionada vulne...  
9  En este caso se instauró la acción de tutela b...

#Exploración inicial

####Análisis exploratorio completo

In [ ]:
#Información general del dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 329 entries, 0 to 328
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   #                329 non-null    int64         
 1   Relevancia       329 non-null    float64       
 2   Providencia      329 non-null    object        
 3   Tipo             0 non-null      float64       
 4   Fecha Sentencia  329 non-null    datetime64[ns]
 5   Tema - subtema   237 non-null    object        
 6   resuelve         329 non-null    object        
 7   sintesis         329 non-null    object        
dtypes: datetime64[ns](1), float64(2), int64(1), object(4)
memory usage: 20.7+ KB


In [ ]:
#Cantidad de valores nulos por columna
df.isnull().sum()

#                    0
Relevancia           0
Providencia          0
Tipo               329
Fecha Sentencia      0
Tema - subtema      92
resuelve             0
sintesis             0
dtype: int64

In [ ]:
df.columns

Index(['#', 'Relevancia', 'Providencia', 'Tipo', 'Fecha Sentencia',
       'Tema - subtema', 'resuelve', 'sintesis'],
      dtype='object')

In [ ]:
df.shape

(329, 8)

In [ ]:
print("Fecha min:",df['Fecha Sentencia'].min())
print("Fecha max:",df['Fecha Sentencia'].max())

Fecha min: 2019-02-01 00:00:00
Fecha max: 2024-11-13 00:00:00


#Observaciones Iniciales

####El DataFrame cuenta con 8 columnas y 329 filas, la columna 'tipo' no contiene información por lo que no aporta información.
####La mayoria de columnas pueden ser cadenas de texto al ser de tipo objeto.